UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs

```diff
@@ -559,21 +561,28 @@ private void SetImageSprite(Sprite value)
 		{
 			spriteRenderer.enabled = true;
 			spriteRenderer.sprite = value;
-			MaterialPropertyBlock block = new MaterialPropertyBlock();
-			spriteRenderer.GetPropertyBlock(block);
-			paletteOrNull = getPaletteOrNull();
-			if (paletteOrNull != null && paletteOrNull.Count == 8)
+			if (isPaletted() && PaletteSet == false)
 			{
-				List<Vector4> pal = paletteOrNull.ConvertAll((c) => new Vector4(c.r, c.g, c.b, c.a));
-				block.SetVectorArray("_ColorPalette", pal);
-				block.SetInt("_IsPaletted", 1);
+				PaletteSet = true;
+				var palette = getPaletteOrNull();
+				if (palette != null && palette.Count == 8)
+				{
+					MaterialPropertyBlock block = new MaterialPropertyBlock();
+					spriteRenderer.GetPropertyBlock(block);
+					List<Vector4> pal = palette.ConvertAll<Vector4>((Color c) => new Vector4(c.r, c.g, c.b, c.a));
```

Comment:
Codacy found an issue: [Remove the type specification; it is redundant.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6392909)